### ISSUE TITLE
Update `seq_linter()` docs to highlight difference between `seq_along()` and `seq_len()`

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2639]
ATM, the docs give no explanation about the difference between these functions and so, to the uninitiated (aka my Python colleagues 😅), it can be a bit confusing as to when to use what.

---

I had in mind something along the following lines. WDYT?

- `seq_along()`: creates a sequence _along_ an object
- `seq_len()`: creates a sequence of a specified _length_

```r
seq_along(iris)
#> [1] 1 2 3 4 5
seq_along(ncol(iris))
#> [1] 1

seq_len(iris)
#> Warning in seq_len(iris): first element used of 'length.out' argument
#> Error in seq_len(iris): argument must be coercible to non-negative integer
seq_len(ncol(iris))
#> [1] 1 2 3 4 5 
```

**Comment by AshesITR** [comment]
What would you suggest to change? Link to the help pages for `seq_{len,along}`?

**Comment by MichaelChirico** [comment]
I'm also not clear what {lintr} should do here, given that we already point to the "correct" alternative in the lint message:

```r
lintr::lint(text = "1:nrow(DF)", linters = lintr::seq_linter())
# <text>:1:1: warning: [seq_linter] 1:nrow(...) is likely to be wrong in the empty edge case. Use seq_len(nrow(...)) instead.
# 1:nrow(DF)
# ^~~~~~~~~~
lintr::lint(text = "1:length(x)", linters = lintr::seq_linter())
# <text>:1:1: warning: [seq_linter] 1:length(...) is likely to be wrong in the empty edge case. Use seq_along(...) instead.
# 1:length(x)
# ^~~~~~~~~~~
```

**Comment by AshesITR** [comment]
While we're at it, might want to rewrite the lint messages to put the appropriate replacement function up first, so it's less easy to miss.

E.g. 

```
Use seq_len(nrow(...)) instead of 1:nrow(...), which is likely to be wrong in the empty edge case.
```

**Comment by MichaelChirico** [comment]
> it really doesn't hurt for {lintr} docs to include one line in the docs to highlight the difference between them.

I just have an aversion to documenting the same thing in many places. It's really hard to keep them in sync in the long run. But happy to review a PR.

**Comment by IndrajeetPatil** [comment]
> While we're at it, might want to rewrite the lint messages to put the appropriate replacement function up first, so it's less easy to miss.
> 
> E.g.
> 
> ```
> Use seq_len(nrow(...)) instead of 1:nrow(...), which is likely to be wrong in the empty edge case.
> ```

Don't we already do this? 

``` r
library(lintr)

lint(
  text = "seq(length(x))",
  linters = seq_linter()
)
#> <text>:1:1: warning: [seq_linter] Use seq_along(...) instead of seq(length(...)), which is likely to be wrong in the empty edge case.
#> seq(length(x))
#> ^~~~~~~~~~~~~~

lint(
  text = "1:nrow(x)",
  linters = seq_linter()
)
#> <text>:1:1: warning: [seq_linter] Use seq_len(nrow(...)) instead of 1:nrow(...), which is likely to be wrong in the empty edge case.
#> 1:nrow(x)
#> ^~~~~~~~~
```

<sup>Created on 2024-08-06 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
that's my fault, I was running from CRAN (actually CRAN-1 I think)

**Comment by AshesITR** [comment]
Oh, sorry for missing that 😅
